exec-agent/docs/ws_api.md

```diff
@@ -0,0 +1,329 @@
+{
+  "method": "process.kill",
+  "id": "id1234567",
+  "params": {
+    "pid": 2
+  }
+}
+```
+
+##### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "id1234567",
+  "result": {
+    "pid": 2,
+    "text": "Successfully killed"
+  }
+}
+```
+
+#### Subscribe to process events
+
+##### Request
+
+- __pid__ - the id of the process to subscribe to
+- __eventTypes__(optional) - comma separated types of events which will be
+received by this channel. By default all the process events will be received
+- __after__(optional) - process logs which appeared after given time will
+be republished to the channel. This parameter may be useful when reconnecting to the exec-agent
+
+```json
+{
+  "method": "process.subscribe",
+  "id": "0x12345",
+  "params": {
+    "pid": 2,
+    "eventTypes": "stdout,stderr",
+    "after" : "2016-07-26T09:36:44.920890113+03:00"
+  }
+}
+```        
+
+
+##### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "0x12345",
+  "result": {
+    "pid": 2,
+    "eventTypes": "stdout,stderr",
+    "text": "Successfully subscribed"
+  }
+}
+```
+
+
+#### Unsubscribe from process events
+
+##### Request
+
+- __pid__ - the id of the process to unsubscribe from
+
+```json
+{
+  "method": "process.unsubscribe",
+  "id": "0x12345",
+  "params": {
+    "pid": 2,
+    "eventTypes": "stdout,stderr",
+    "after": "2016-07-26T09:36:44.920890113+03:00"
+  }
+}
+```
+
+##### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "0x12345",
+  "result": {
+    "pid": 2,
+    "text": "Successfully unsubscribed"
+  }
+}
+```
+
+#### Update process subscriber
+
+##### Request
+
+- __pid__ - the id of the process which subscriber should be updated
+- __eventTypes__ - comma separated types of events which will be
+received by this channel.
+
+```json
+{
+  "method": "process.updateSubscriber",
+  "id": "0x12345",
+  "params": {
+    "pid": 2,
+    "eventTypes": "process_status"
+  }
+}
+```
+
+##### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "0x12345",
+  "result": {
+    "pid": 2,
+    "eventTypes": "process_status",
+    "text": "Subscriber successfully updated"
+  }
+}
+```
+
+#### Get process logs
+
+##### Request
+
+- __pid__ - the id of the process to get logs
+- __from__(optional) - time to get logs from e.g. _2016-07-12T01:48:04.097980475+03:00_ the format is _RFC3339Nano_
+- __till__(optional) - time to get logs till e.g. _2016-07-12T01:49:04.097980475+03:00_ the format is _RFC3339Nano_
+- __format__(optional) - the format of the response, default is `json`, possible values are: `text`, `json`
+- __limit__(optional) - the limit of logs in result, the default value is _50_, logs are limited from the
+latest to the earliest
+- __skip__ (optional) - the logs to skip, default value is `0`
+
+```json
+{
+  "method": "process.getLogs",
+  "id": "0x12345",
+  "params": {
+    "pid": 3,
+    "limit": 5,
+    "skip": 5
+  }
+}
+```
+
+##### Response
+
+For the command `printf "1\n2\n3\n4\n5\n6\n7\n8\n9\n10`, the result will look like
```

Comment:
Request section for this response covers your questions.
> Does it show only logs produced before this request got to the server?

It will use logs `[first logged message, time.Now()]` where `time.Now` is a time when server receives request